roxie/roxiemem/roxiemem.hpp

```diff
@@ -232,7 +232,7 @@ class roxiemem_decl DataBufferBottom : public HeapletBase
 private:
     friend class CDataBufferManager;
     CDataBufferManager * volatile owner;
-    atomic_t okToFree;
+    std::atomic_uint okToFree;      // use uint since it is more efficient on some architectures
```

Comment:
Yes, that might be better.  The only time it isn't compatible is if you use compare_exchange.